### DISCUSSION TITLE
What amount of storage is required for downloads/omfile data storage?

### DISCUSSION THREAD

**Discussion by delphir-com** [open-meteo/open-meteo#384] [Q&A] [answered]
What amount of space is required for downloads/omfile data storage?

I saw in the docs that it is ~150GB, but I was trying to download only temperature/wind/pressure/(and a bit) variables, and I`m already over 151GB.

The reason I`m asking is because I see discrepancies in the results returned from self-hosted and public open-meteo api. From the code it looks like many variables rely on "other" variables, so if downloader is loading just some of them - then calculations may be incorrect. So I would like to try a "full" data download, but just wondering what amount of storage is required for all downloads/processed files?

Related side-question: is it safe to delete `./data/download-*` folder after a certain downloader has finished it's work? or previously downloaded files are used in future runs too?

**Comment by patrick-zippenfenig** [comment, marked as answer]
~150 GB is really the minimum amount of storage for high resolution models and a smaller selection of variables. Including all variables, it is around ~350 GB. With 4-6 months of past data, it adds up to around 800 GB.

Not including any ensemble member or historical weather data. Ensemble forecasts quickly add another 400 GB.

Yes, you can delete `download-*` directories after download. However, the next download, will overwrite previously downloaded data and the amount of storage remains the same.

**Comment by delphir-com** [comment]
Got it, please consider to update the documentation with details provided in your answer.

https://github.com/open-meteo/open-meteo/blob/e24d50aee0d0c5c7107519e84bf67531bb814a3e/docs/getting-started.md?plain=1#L16

**Comment by delphir-com** [comment]
whats the approx amount of storage needed just for the `omfile-*` folders (I assume this is where the actual data is stored used for forecasts)?

**Reply by patrick-zippenfenig** [reply]
Actually I was only referring to the `omfile-*` directories ;-)

**Comment by delphir-com** [comment]
what's the difference between `past data` and `historical data`? as you mentioned them separately...
if the past/historical-data is participating in forecasts (e.g. forecasts starting from today 00:00) calculations or it can be omitted? if so - then which downloaders are responsible for it?

also from what I understand - `ensemble` is kinda an important part for the forecasts as it adds a correction/probability, right?

**Reply by patrick-zippenfenig** [reply]
`Past data` is old data from the forecast APIs. With historical data I referred to data for the historical weather API which is multiple TB of data.

Ensemble forecasts do not yet influence the main weather forecast. In the future, I will derive some variables like for certain a forecast is or more details about precipitation probability, but right now, it is mostly separated

**Comment by dzungpv** [comment]
Seem old data not auto delete, It just there. I tested with 300GB and 2 days full off download file, the data is only 37 GB

**Reply by patrick-zippenfenig** [reply]
All downloaders maintain the same amount of disk space. Initially the download directories might be larger than the actually `omfile-` database, because the compression is less efficient in this case. Of course the `omfile-` database will continue to grow to create an archive of past weather data

**Reply by dzungpv** [reply]
After 2 week running without "Ensemble models" and "cams_global" my storage for data folder is 350GB. Is there anyway to config how long past data keep? For example I only want to keep 14 days of the data in all om files for all models.
